pkg/lint/lint_test.go

```diff
@@ -205,10 +227,28 @@ func TestGoModLint(t *testing.T) {
 	}
 
 	for _, modFile := range modFiles {
+		assert.Equalf(t, runtime.Version()[2:], modFile.Go.Version, "%s's go directive should be match runtime.Version(): %q", modFile.Module.Mod.Path, runtime.Version()[2:])
+
 		for _, r := range modFile.Require {
 			if strings.HasPrefix(r.Mod.Path, modPrefix) {
 				assertPresentInMainOrRelease(t, modFile, r.Mod)
 			}
+
+			if version, ok := workspaceWideVersions[r.Mod.Path]; ok {
+				assert.Equalf(t, version, r.Mod.Version, `
+%s MUST be %s across the all go.mod's in this repository.
+
+	%s is using %s
+
+Quick Fixes:
+
+	go get %s@%s
+	go mod edit -require=%s@%s
+	%s %s
```

Comment:
What this line mean when the assertion is not equal?

Reply:
It just prints out `module version` so you can copy and paste it into the require block of go.mod. It's not very pretty to read but the output is at least useable.